### PR TITLE
Entity confusion matrix

### PR DESCRIPTION
**Pull request by SoluMilken**
hmm... I don't know how to write unittest for plotting. :scream:

### CODE REVIEW

**Comment by SoluMilken** [comment]
```
$ flake8
./ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py:43:35: E999 SyntaxError: invalid syntax
1     E999 SyntaxError: invalid syntax
```
./ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py:43:35: C815 missing trailing comma in Python 3.5+
1     C815 missing trailing comma in Python 3.5+

```
```

File: ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                    ),
+                    ['1', '2', '3', '4', '5', '6'],
+                ),
+            ),
+        ]
+        for i, test_case in enumerate(test_cases):
+            with self.subTest(i=i):
+                result = entity_confusion_matrix(
+                    **test_case[0],
+                )
```
Comment (stegben) [inline ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py]: maybe put it in a single line?

**Comment by stegben** [comment]
Think we can skip the CI on 3.7 until it publicly release

**Comment by SoluMilken** [comment]
Let's skip python 3.7 !!!

**Comment by stegben** [comment]
skiped in #32

**Comment by SoluMilken** [comment]
@GBLin5566 We need your review. !! :joy:

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,8 @@
 gql-fork==0.2.0
+scipy
+numpy
 requests==2.13.0
 flake8==3.5.0
 flake8-config-yoctol==0.0.9
+matplotlib
+sklearn
```
Comment (stegben) [inline requirements.txt]: add these packages into `setup.py`

File: ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+from unittest import TestCase
+
+import numpy as np
+
+from ..entity_confusion_matrix import (
+    entity_confusion_matrix,
+    # entity_confusion_matrix_figure,
```
Comment (GBLin5566) [inline ynlu/sdk/evaluation/tests/test_entity_confusion_matrix.py]: 這個figure是什麼R

File: ynlu/sdk/evaluation/entity_confusion_matrix.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        range(confusion_matrix.shape[1]),
+    ):
+        plt.text(j, i, format(confusion_matrix[i, j], fmt),
+                 horizontalalignment="center",
+                 color="white" if confusion_matrix[i, j] > thresh else "black")
+
+    plt.tight_layout()
+    plt.ylabel('True Label')
+    plt.xlabel('Predicted Label')
+    plt.show()
```
Comment (GBLin5566) [inline ynlu/sdk/evaluation/entity_confusion_matrix.py]: 應該要有個機制是可以選擇要save還是show？

Comment (GBLin5566) [reply ynlu/sdk/evaluation/entity_confusion_matrix.py]: 可以傳 path = None 進這個function，
如果是None就plt.show()，不是就用那個path plt.save()

Comment (SoluMilken) [reply ynlu/sdk/evaluation/entity_confusion_matrix.py]: 0da1124

File: requirements.txt (hunk 2 above)
Comment (SoluMilken) [reply requirements.txt]: 9cad12a

File: ynlu/sdk/evaluation/entity_confusion_matrix.py (hunk 4 above)
Comment (SoluMilken) [reply ynlu/sdk/evaluation/entity_confusion_matrix.py]: 喔喔喔喔喔喔喔喔喔喔喔喔喔喔

Comment (SoluMilken) [reply ynlu/sdk/evaluation/entity_confusion_matrix.py]: 4569f6b

**Comment by SoluMilken** [comment]
????哪兩個不用 RRR

**Comment by stegben** [comment]
Oh I mean `flake8` and `flake8-config-yoctol`

**Pull request merged**